### ISSUE TITLE
SAML2 Authentication Request Signature rejected

### ISSUE THREAD

**Issue by lemoinem** [wso2/product-is#9196]
**Describe the issue:**
WSO2 rejects SAML2 Authentication Request Signature while external tool (https://www.samltool.com/validate_authn_req.php) confirms the Signature is valid.

***Example request:***

URL: __badly encoded URL__

```
**How to reproduce:**
1. Configure SAML2 SSO SP in IS. Use attached Certificate, Enable Signature Validation in Authentication Requests and Logout Requests, Enable SAML2 Artifact Binding, Set signature algorithm to `http://www.w3.org/2001/04/xmldsig-more#rsa-sha256`
2. Try to login using the above SAML Authentication Request.
3. WSO2 IS rejects the request saying the signature is invalid, although external tool confirms the signature is valid.

Note:
1. Disabling "Signature Validation in Authentication Requests and Logout Requests" will make IS accept the request.
2. Adding the certificate to wso2carbon.jks and using either wso2carbon or the imported certificate itself as alias doesn't change the behavior.
3. Using HttpPost binding to send the request instead of HttpRedirect will make IS accept the request (therefore the certificate is correctly configured)

**Expected behavior:**
The signature being valid (as confirmed by an independent tool), it should be accepted.

**Environment information :**
 - Product Version: IS 5.10.0
 - OS: Windows Server 2012 R2
 - Database: H2

**Comment by lemoinem** [comment]
After further investigation, it seems there was an issue with how the HTTP Redirect URL was processed on my end.
Something about the URL escaping was lost and caused the signature to not match anymore.
Once I fixed this, WSO2 IS accepted the signed requests without further issue.

I have removed sensitive info from the original issue description.
However, since it will linger in the edit history, may I ask you ignore and delete this issue? (I am keeping the issue opened mostly so the deletion request is seen).

This behavior from WSO2 could still be "fixed". For example, by ensuring WSO2 IS URL encodes properly each part of the URL, as specified in the standard... But I can see how changing that would also be a breaking change (and arguably a worst behavior as it would prevent tolerating SP that URL encode their data slightly differently than described in the standard, but are still internally consistent).

Sorry for the inconvenience.
